src/replace.js

```diff
@@ -14,9 +14,7 @@ import { setByPath } from "./util.js";
  */
 export default function replace (child, newChild) {
 	const parentPath = getPath.call(this, child);
-	if (parentPath === undefined) {
-		throw new Error("Cannot replace a child node with no parent pointer. Call setParent() on the node or updateParents() on an ancestor to add parent pointers to this node");
-	}
+	checkParentPointers.call(this, child);
```

Comment:
So here by making a change to make our codebase more neat, we ended up making the user experience worse, which is an antipattern. The original error was actually more clear. Remember that internal complexity should be preferred over external (user-facing) complexity. Perhaps we can get the best of both worlds by passing the message in, but then it's unclear what benefit the function actually gets us. Perhaps the message could be the part that changes ("Cannot replace a child node with no parent pointer.") and the rest can be added automatically? But does the rest work with every case?

Reply:
This is currently the only place in our codebase throwing an error, so I think we can do something like `${customMessage} Call setParent() on the node or updateParents() on an ancestor to add parent pointers to this node` to have the last part be consistent

Reply:
Sounds good. Make sure to cover the case where no message is set.